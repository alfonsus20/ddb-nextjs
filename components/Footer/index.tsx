import { Box, Button, Flex, Input, Text, Textarea, VStack } from "@chakra-ui/react";
import React from "react";

const Footer = () => {
  return (
    <Box bgColor="red.600" color="white">
      <Flex maxW="container.xl" mx="auto" py={12}>
        <Box w="33.33%">
          <Text fontSize="2xl" fontWeight='semibold' mb={4}>
            Data Mahasiswa
          </Text>
          <Box>
            <Text>Data Alumni</Text>
            <Text>Data Mahasiswa Aktif</Text>
          </Box>
        </Box>
        <Box w="33.33%">
          <Text fontSize="2xl" fontWeight='semibold' mb={4}>
            Sosial Media
          </Text>
          <Box>
            <Text>Instagram</Text>
            <Text>Youtube</Text>
          </Box>
        </Box>
        <Box w="33.33%">
          <Text fontSize="2xl" fontWeight='semibold' mb={4}>
            Hubungi Kami
          </Text>
          <VStack as="form" spacing={4} align="flex-start">
            <Input placeholder="Nama" _placeholder={{color : 'white'}} />
            <Textarea placeholder="Pesan"  _placeholder={{color : 'white'}} resize='none' />
            <Button colorScheme="whiteAlpha">Kirim</Button>
          </VStack>
        </Box>
      </Flex>
      <Box as="footer" textAlign="center" py={4} bgColor="#333333">
        &copy; Dara Daeng Brawijaya 2022. All Rights Reserved.
      </Box>
    </Box>
  );
};

export default Footer;
