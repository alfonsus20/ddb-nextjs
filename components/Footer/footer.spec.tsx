import { fireEvent, render } from "@testing-library/react";
import Footer from ".";

describe("Footer unit test", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match snapshot", () => {
    const container = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it("should be able to handle form change", () => {
    const { getByTestId } = render(<Footer />);

    const subjectInput = getByTestId("subject-input") as HTMLInputElement;
    const bodyInput = getByTestId("body-input") as HTMLTextAreaElement;

    expect(subjectInput).toBeVisible();
    expect(bodyInput).toBeVisible();

    fireEvent.change(subjectInput, { target: { value: "Judul" } });
    expect(subjectInput.value).toEqual("Judul");

    fireEvent.change(bodyInput, { target: { value: "Body" } });
    expect(bodyInput.value).toEqual("Body");
  });

  it("should call window.open after submitting form when fields are complete", () => {
    const { getByTestId } = render(<Footer />);

    const subjectInput = getByTestId("subject-input") as HTMLInputElement;
    const bodyInput = getByTestId("body-input") as HTMLTextAreaElement;

    fireEvent.change(subjectInput, { target: { value: "Judul" } });
    fireEvent.change(bodyInput, { target: { value: "Body" } });

    const btnSubmit = getByTestId("btn-submit") as HTMLButtonElement;
    fireEvent.click(btnSubmit);

    expect(global.open).toHaveBeenCalledWith(
      "mailto:ddbrawijaya@gmail.com?subject=Judul&body=Body"
    );
  });

  it("should not call window.open after submitting form when fields are not complete", () => {
    const { getByTestId } = render(<Footer />);

    const subjectInput = getByTestId("subject-input") as HTMLInputElement;

    fireEvent.change(subjectInput, { target: { value: "Judul" } });

    const btnSubmit = getByTestId("btn-submit") as HTMLButtonElement;
    fireEvent.click(btnSubmit);

    expect(global.open).not.toHaveBeenCalled();
  });
});
