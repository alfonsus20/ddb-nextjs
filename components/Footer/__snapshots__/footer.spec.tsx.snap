// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer unit test should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-1b98ggy"
      >
        <div
          class="css-1r1d5mt"
        >
          <div
            class="css-5dpfxv"
          >
            <p
              class="chakra-text css-y50o8b"
            >
              Data Mahasiswa
            </p>
            <div
              class="css-0"
            >
              <div
                class="css-1x7yhvo"
                href="/alumni"
              >
                Data Alumni
              </div>
              <div
                class="css-1x7yhvo"
                href="/mahasiswa"
              >
                Data Mahasiswa Aktif
              </div>
            </div>
          </div>
          <div
            class="css-5dpfxv"
          >
            <p
              class="chakra-text css-y50o8b"
            >
              Sosial Media
            </p>
            <div
              class="css-1ezsviu"
            >
              <a
                class="css-1trwvm"
                href="https://www.instagram.com"
                rel="noopen noreferrer"
                target="_blank"
              >
                <div
                  class="css-17gsuhn"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                    />
                  </svg>
                  Instagram
                </div>
              </a>
              <a
                class="css-1trwvm"
                href="https://www.instagram.com"
                rel="noopen noreferrer"
                target="_blank"
              >
                <div
                  class="css-17gsuhn"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                    />
                  </svg>
                  Youtube
                </div>
              </a>
            </div>
          </div>
          <div
            class="css-1akz1q2"
          >
            <p
              class="chakra-text css-y50o8b"
            >
              Hubungi Kami
            </p>
            <form
              action="mailto:ddbrawijaya@gmail.com"
              class="chakra-stack css-153jur6"
              method="GET"
            >
              <input
                aria-required="true"
                class="chakra-input css-1r3z6tj"
                data-testid="subject-input"
                name="subject"
                placeholder="Subjek"
                required=""
                value=""
              />
              <textarea
                aria-required="true"
                class="chakra-textarea css-cur9v4"
                data-testid="body-input"
                name="body"
                placeholder="Pesan"
                required=""
              />
              <button
                class="chakra-button css-5c2176"
                data-testid="btn-submit"
                type="submit"
              >
                Kirim
              </button>
            </form>
          </div>
        </div>
        <footer
          class="css-gjn7b"
        >
          Â© Dara Daeng Brawijaya 2022. All Rights Reserved.
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1b98ggy"
    >
      <div
        class="css-1r1d5mt"
      >
        <div
          class="css-5dpfxv"
        >
          <p
            class="chakra-text css-y50o8b"
          >
            Data Mahasiswa
          </p>
          <div
            class="css-0"
          >
            <div
              class="css-1x7yhvo"
              href="/alumni"
            >
              Data Alumni
            </div>
            <div
              class="css-1x7yhvo"
              href="/mahasiswa"
            >
              Data Mahasiswa Aktif
            </div>
          </div>
        </div>
        <div
          class="css-5dpfxv"
        >
          <p
            class="chakra-text css-y50o8b"
          >
            Sosial Media
          </p>
          <div
            class="css-1ezsviu"
          >
            <a
              class="css-1trwvm"
              href="https://www.instagram.com"
              rel="noopen noreferrer"
              target="_blank"
            >
              <div
                class="css-17gsuhn"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                  />
                </svg>
                Instagram
              </div>
            </a>
            <a
              class="css-1trwvm"
              href="https://www.instagram.com"
              rel="noopen noreferrer"
              target="_blank"
            >
              <div
                class="css-17gsuhn"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                  />
                </svg>
                Youtube
              </div>
            </a>
          </div>
        </div>
        <div
          class="css-1akz1q2"
        >
          <p
            class="chakra-text css-y50o8b"
          >
            Hubungi Kami
          </p>
          <form
            action="mailto:ddbrawijaya@gmail.com"
            class="chakra-stack css-153jur6"
            method="GET"
          >
            <input
              aria-required="true"
              class="chakra-input css-1r3z6tj"
              data-testid="subject-input"
              name="subject"
              placeholder="Subjek"
              required=""
              value=""
            />
            <textarea
              aria-required="true"
              class="chakra-textarea css-cur9v4"
              data-testid="body-input"
              name="body"
              placeholder="Pesan"
              required=""
            />
            <button
              class="chakra-button css-5c2176"
              data-testid="btn-submit"
              type="submit"
            >
              Kirim
            </button>
          </form>
        </div>
      </div>
      <footer
        class="css-gjn7b"
      >
        Â© Dara Daeng Brawijaya 2022. All Rights Reserved.
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
